main.c

```diff
@@ -235,8 +235,41 @@ on_check_config(TCMUService1 *interface,
 	return TRUE;
 }
 
+static gboolean
+on_mediachange(TCMUService1 *interface,
+	       GDBusMethodInvocation *invocation,
+	       guint64 size,
+	       gchar *cfgstring,
+	       gpointer user_data)
+{
+	struct tcmur_handler *handler = user_data;
+	struct tcmu_device *dev = handler->dev;
+	char *reason = NULL;
+	int ret = 0;
+
+	tcmu_set_dev_cfgstring(dev, cfgstring);
+	tcmu_update_num_lbas(dev, size);
+	tcmu_set_dev_size(dev);
+
+	tcmu_block_device(dev);
+	tcmu_flush_device(dev);
+	ret = tcmu_reopen_dev(dev);
+	if (ret == 0)
+		reason = "success";
+
+	tcmu_unblock_device(dev);
+	tcmur_set_pending_ua(dev, TCMUR_UA_DEV_MEDIUM_CHANGED);
+	g_dbus_method_invocation_return_value(invocation,
+		g_variant_new("(is)", ret, reason ? : "unknown"));
+
+	if (ret != 0)
+		free(reason);
```

Comment:
Does dbus allocate memory for "reason", or was this leftover from another version of the code.

Reply:
I will remove the free, I took it from check_config. They used it because they used strdup which handles the malloc in the background, and I dont use that, so I can remove.